src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -327,8 +318,20 @@ public void postBuildStatus(@NonNull BitbucketBuildStatus status) throws IOExcep
      * {@inheritDoc}
      */
     @Override
-    public boolean checkPathExists(@NonNull String branch, @NonNull String path) throws IOException {
-        return HttpStatus.SC_OK == getRequestStatus(String.format(API_BROWSE_PATH, getUserCentricOwner(), repositoryName, path, branch));
+    public boolean checkPathExists(@NonNull String branchOrHash, @NonNull String path) throws IOException {
+        StringBuilder encodedPath = new StringBuilder(path.length() + 10);
+        boolean first = true;
+        for (String segment : StringUtils.split(path, "/")) {
```

Comment:
This is another bug I found when reviewing the code to ensure that the fix for JENKINS-43664 prototyped in https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/50 was present in this PR. (#50 has bugs in that it uses the wrong encoder for some things... which had me look at all the encoding... and then I found that this was wrong... likely not spotted before as most people are just looking for the path `Jenkinsfile` which doesn't have any `/` or space in it